### PR TITLE
Fixed element type destruction in global scope

### PR DESCRIPTION
**Pull request by apankratovantonp**
### Details:
 - Static map with element type info can be destroyed before static object with element type get type info.

### CODE REVIEW

File: src/core/src/type/element_type.cpp [hunk 1]
```diff
@@ -109,7 +105,28 @@ ov::element::Type::Type(size_t bitwidth,
                         bool is_signed,
                         bool is_quantized,
                         const std::string& /* cname */) {
-    for (const auto& t : get_type_info_map()) {
+    const ElementTypes::ElementsMap elements_map{
```
Comment (nosovmik) [inline src/core/src/type/element_type.cpp:108]: this is quite unusual constructor which seems not covered by existing unit tests.

Comment (apankratovantonp) [reply src/core/src/type/element_type.cpp:108]: @ilyachur can it be removed?

File: src/core/include/openvino/core/type/element_type.hpp [hunk 2]
```diff
@@ -63,7 +63,7 @@ class OPENVINO_API Type {
     constexpr Type(const Type_t t) : m_type{t} {}
     Type(size_t bitwidth, bool is_real, bool is_signed, bool is_quantized, const std::string& cname);
     Type& operator=(const Type&) = default;
-    const std::string& c_type_string() const;
+    std::string c_type_string() const;
```
[the remark this answers is not in the document]
Comment (apankratovantonp) [reply src/core/include/openvino/core/type/element_type.hpp:66]: Yes, const reference extend lifetime of temporary object.

File: src/core/src/type/element_type.cpp (hunk 1 above)
Comment (ilyachur) [reply src/core/src/type/element_type.cpp:108]: Do you mean remove the constructor? No we cannot. But we can deprecate it and remove in the furure

File: src/core/include/openvino/core/type/element_type.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (ilyachur) [reply src/core/include/openvino/core/type/element_type.hpp:66]: I don't believe tthat we have customers which use this method. I am ok with such changes.

**Review by ilyachur — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Pull request merged**